Backport query rule alert suppression to API docs (#4496)
elastic/security-docs: docs/detections/api/rules/rules-api-create.asciidoc

|==============================================

[[opt-fields-alert-suppression-create]]
===== Optional alert suppression fields for query rules

preview::[]

[width="100%",options="header"]
|==============================================
|Name |Type |Description

|alert_suppression |Object a|Defines alert suppression configuration. Available fields:

* `group_by` (string[], required): An array of 1-3 field names to use for suppressing alerts.

> * `duration` (<<alert-suppression-duration-schema, duration object>>, optional): Defines the time period in which alerts will be suppressed. If absent, alerts will be suppressed on rule execution only.

Reviewer: See https://github.com/elastic/security-docs/pull/4479/files#r1434176943

Author: ```suggestion
* `duration` (<<alert-suppression-duration-schema, duration object>>, optional): The time period in which alerts will be suppressed, beginning when the rule first meets its criteria and creates the alert. If not specified, alerts will be suppressed on rule execution only.
```

> * `missing_fields_strategy` (string, optional): Defines how to handle events with missing suppression fields. Values:

Reviewer: ```suggestion
* `missing_fields_strategy` (string, optional): Defines how to handle events with missing suppression fields. Possible values:
```

NOTE: Only threats described using the MITRE ATT&CK^TM^ framework are displayed
in the UI (*Rules* -> *Detection rules (SIEM)* -> *_Rule name_*).

[[alert-suppression-duration-schema]]
===== `alert_suppression.duration` schema

All fields are required:

[width="100%",options="header"]
|==============================================
|Name |Type |Description

> |unit |string | Time unit, takes value of `s`(seconds), `m`(minutes), `h`(hours).

Reviewer: ```suggestion
|unit |string | Time unit. Possible values are: `s`(seconds), `m`(minutes), or `h`(hours).
```

> |value |number | Positive number

Reviewer: ```suggestion
|value |number | Positive number.
```

}
--------------------------------------------------

*Example 8*

> Query rule that searches for processes started by MS Office and suppress alerts by `process.parent.name` field on 5 hours time period:

Reviewer: ```suggestion
Query rule that searches for processes started by MS Office and suppresses alerts by the `process.parent.name` field within a 5-hour time period:
```
